### DISCUSSION TITLE
Force the use of H3 (TLSv1.3) over DoH (DNS over Https)

### DISCUSSION THREAD

**Discussion by junzhli** [DNSCrypt/dnscrypt-proxy#2644] [Ideas]
Since DoQ (DNS over Quic) is not supported at this moment, is it possible to explicitly force it to use H3 protocol in supported remote resolvers? Thanks!

**Comment by jedisct1** [comment]
Set `http3 = true` in the configuration file. HTTP/3 will be used if the server supports it.

**Reply by junzhli** [reply]
Seems like it can only handle Doh in HTTP/3 when there's a header `src-alt` containing `h3` magic string.
I tried with `[unique-id].dns.nextdns.io` and it didn't work due to the above reason and ended up with using `dot3.dns.nextdns.io/[unique-id]`, which hits the above condition.
Not sure if the issue is from NextDNS, anyways many thanks! @jedisct1

**Reply by zubrRB** [reply]
Having a problem with this when using AdGuard. https://github.com/AdguardTeam/CoreLibs/issues/1892

**Comment by junzhli** [comment]
Hi @jedisct1 @zubrRB 

I made some changes on this to force the use of httpv3 on the supplied list of hosts
It helps a lot in my case and hopefully it can be part of feature in release version!

https://github.com/DNSCrypt/dnscrypt-proxy/pull/2846

**Comment by bcookatpcsd** [comment]
I had to enable http3 as well as the probe for http3

Now I see this in the logs: 
![image](https://github.com/user-attachments/assets/dd50c0e0-6e89-4977-b806-2effcda52017)

and via tcpdump

```
10:51:48.128488 IP (tos 0x2,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 248)
    10.120.48.253.8235 > 45.90.28.0.443: UDP, length 220
10:51:48.202709 IP (tos 0x2,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 249)
    10.120.48.253.8235 > 45.90.28.0.443: UDP, length 221
10:51:48.244496 IP (tos 0x2,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 62)
    10.120.48.253.8235 > 45.90.28.0.443: UDP, length 34
10:51:48.273566 IP (tos 0x2,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 62)
    10.120.48.253.8235 > 45.90.28.0.443: UDP, length 34
10:51:48.278660 IP (tos 0x2,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 315)
    10.120.48.253.8235 > 45.90.28.0.443: UDP, length 287
10:51:48.290223 IP (tos 0x2,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 61)
    10.120.48.253.8235 > 45.90.28.0.443: UDP, length 33
10:51:48.293601 IP (tos 0x2,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 70)
    10.120.48.253.8235 > 45.90.28.0.443: UDP, length 42
10:51:48.297853 IP (tos 0x2,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 61)
    10.120.48.253.8235 > 45.90.28.0.443: UDP, length 33
10:51:48.326999 IP (tos 0x2,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto UDP (17), length 62)
    10.120.48.253.8235 > 45.90.28.0.443: UDP, length 34
```

```
 dnscrypt-proxy -version
2.1.12
```
